Modules/_cursesmodule.c

```diff
@@ -3255,6 +3255,67 @@ _curses_setupterm_impl(PyObject *module, const char *term, int fd)
     Py_RETURN_NONE;
 }
 
+/*[clinic input]
+_curses.set_escdelay
+    ms: int
+        length of the delay in milliseconds.
+    /
+
+Sets the curses ESCDELAY setting.
+
+Sets the number of milliseconds to wait after reading an escape character,
+to distinguish between an individual escape character entered on the
+keyboard from escape sequences sent by cursor and function keys.
+[clinic start generated code]*/
+
+static PyObject *
+_curses_set_escdelay_impl(PyObject *module, int ms)
+/*[clinic end generated code: output=43818efbf7980ac4 input=7796fe19f111e250]*/
+{
+    if (ms <= 0) {
+        PyErr_SetString(PyExc_ValueError, "ms must be > 0");
+        return NULL;
+    }
+
+    int code = set_escdelay(ms);
+    if (code != ERR) {
```

Comment:
You can use `PyCursesCheckERR()`.

Reply:
ah, I copied this from elsewhere in the file -- should I fix those as well?

Reply:
It is better to do this in a separate tiny PR (no issue is needed).

Reply:
awesome, done!